Detectors/ITSMFT/ITS/reconstruction/src/CookedTracker.cxx

```diff
@@ -477,9 +481,37 @@ void CookedTracker::process(gsl::span<const Cluster> const& clusters, TrackInser
 
   auto start = std::chrono::system_clock::now();
 
-  mFirstCluster = &clusters[0];
+  mFirstInFrame = rof.getFirstEntry();
+
+  mClusterCache.reserve(rof.getNEntries());
+  auto clusters_in_frame = rof.getROFData(clusters);
+  for (const auto& comp : clusters_in_frame) {
+
+    auto pattID = comp.getPatternID();
+    Point3D<float> locXYZ;
+    float sigmaY2 = 0.0015 * 0.0015, sigmaZ2 = sigmaY2, sigmaYZ = 0; //Dummy COG errors (about half pixel size)
+    if (pattID == itsmft::CompCluster::InvalidPatternID || dict.IsGroup(pattID)) {
+      o2::itsmft::ClusterPattern patt(pattIt);
+      locXYZ = dict.getClusterCoordinates(comp, patt);
```

Comment:
Why not getting the errors from dictionary also for the groups? I think the case of `InvalidPatternID` should be treated separately (and as the last `if` case: the priority should be unique_IDs then groups then InvalidPatternID).

Concerning dict.getClusterCoordinates usage: it calls `TopologyDictionary::GetXcog`, which, as the whole bunch of methods in https://github.com/AliceO2Group/AliceO2/blob/cf5bf4b122e5ec9e02777cf246c3fa91d5033f62/DataFormats/Detectors/ITSMFT/common/src/TopologyDictionary.cxx#L105-L177
will throw an exception if specific dictionary was not loaded (e.g. in the "1st" production used to create a dictionary). 

@lbariogl could you please change all these checks ``if (n < 0 || n >= (int)mVectorOfGroupIDs.size()) ..`` to `assert(n<(int)mVectorOfGroupIDs.size())`, so that it does not affect performance in the production?
Also, I would inline all these methods.

Reply:
Ok @shahor02, I will do it in a separate pull request

Reply:
@shahor02 Under normal conditions, this throw should never happen. For example, during the "1st production", the execution flow will go via the "InvalidPatternID" branch of this if.

Reply:
right about the exception, but why mixing InvalidPatternID and groups and assigning a dummy error also to groups?